### ISSUE TITLE
logrotated rev 9 fails to install on bionic

### ISSUE THREAD

**Issue by jneo8** [canonical/charm-logrotated#4]
Hi team! 

I'm deploying latest/stable logrotated (rev 9) on the lma model with nagios latest/stable rev 59. And the install hook fails with the following conflict:

```
root@nagios-1:/var/lib/juju/agents/unit-logrotated-0/charm# ./hooks/install
Reading package lists... Done
Building dependency tree
Reading state information... Done
build-essential is already the newest version (12.4ubuntu1).
python3-yaml is already the newest version (3.12-1build2).
python3-dev is already the newest version (3.6.7-1~18.04).
python3-setuptools is already the newest version (39.0.1-2ubuntu0.1).
python3-wheel is already the newest version (0.30.0-0.2ubuntu0.1).
python3-pip is already the newest version (9.0.1-2.3~ubuntu1.18.04.8+esm1).
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
Reading package lists... Done
Building dependency tree
Reading state information... Done
0 upgraded, 0 newly installed, 0 to remove and 1 not upgraded.
The directory '/home/ubuntu/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/ubuntu/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting pip
pip requires Python '>=3.7' but the running Python is 3.6.9
Traceback (most recent call last):
  File "./hooks/install", line 8, in <module>
    basic.bootstrap_charm_deps()
  File "lib/charms/layer/basic.py", line 202, in bootstrap_charm_deps
    _update_if_newer(pip, pre_install_pkgs)
  File "lib/charms/layer/basic.py", line 367, in _update_if_newer
    pkg], env=_get_subprocess_env())
  File "/usr/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/var/lib/juju/agents/unit-logrotated-0/.venv/bin/pip', 'install', '-U', '--no-index', '-f', 'wheelhouse', 'pip']' returned non-zero exit status 1.
```

I have logrotated working fine with the other lma units and the previous revision (14, which is now in the latest/candidate) was also working correctly.

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2023-05-18T12:00:46Z
 * owner: [natalytvinova](https://launchpad.net/~natalytvinova)
 * assignee: [None](https://launchpad.net/~None)
 
 
 * the launchpad [url](https://bugs.launchpad.net/bugs/2020081)

**Comment by jneo8** [comment]
(by [aieri](https://launchpad.net/~aieri))
This is a known issue that was fixed in commit 26a61a195291422cc8eead5495fcf73613402a32 (and rev 14).

**Comment by jneo8** [comment]
(by [gabrielcocenza](https://launchpad.net/~gabrielcocenza))
For some unknown reason, the stable release was changed today in a revision that didn't have the patch for bionic compatibility. I've changed again the channel stable to rev 14 and the problem is solved.
